tools/nodejs_api/src_cpp/node_database.cpp

```diff
@@ -0,0 +1,69 @@
+#include "node_database.h"
+
+#include "main/kuzu.h"
+
+using namespace kuzu::main;
+
+Napi::FunctionReference NodeDatabase::constructor;
+
+Napi::Object NodeDatabase::Init(Napi::Env env, Napi::Object exports) {
+    Napi::HandleScope scope(env);
+
+    Napi::Function t = DefineClass(env, "NodeDatabase", {
+          InstanceMethod("resizeBufferManager", &NodeDatabase::ResizeBufferManager),
+      });
+
+    constructor = Napi::Persistent(t);
+    constructor.SuppressDestruct();
+
+    exports.Set("NodeDatabase", t);
+    return exports;
+}
+
+
+NodeDatabase::NodeDatabase(const Napi::CallbackInfo& info) : Napi::ObjectWrap<NodeDatabase>(info)  {
+    Napi::Env env = info.Env();
+    Napi::HandleScope scope(env);
+
+    if (info.Length()!=2) {
```

Comment:
The buffer manager size should be optional. If it is not provided, setting it to 0 so that the system can automatically configure it: https://github.com/kuzudb/kuzu/blob/89fba7c1328b543747fc4948e6bff0961263276b/tools/python_api/src_cpp/py_database.cpp#L6

Reply:
> ` constructor(databaseConfigString, bufferSize = 0) {
        this.database = new kuzu.NodeDatabase(databaseConfigString, bufferSize);
    }`
Is it fine if I just do this in the javascript side -> if a user doesn't specifiy an argument then it will pass 0 to the NodeDatabase class

Reply:
You can do it on JS side